crates/katana/primitives/src/utils/transaction.rs

```diff
@@ -231,3 +334,37 @@ mod tests {
         );
     }
 }
+
+fn encode_gas_bound(name: &[u8], bound: &ResourceBounds) -> FieldElement {
+    let mut buffer = [0u8; 32];
+    let (remainder, max_price) = buffer.split_at_mut(128 / 8);
+    let (gas_kind, max_amount) = remainder.split_at_mut(64 / 8);
+
+    let padding = gas_kind.len() - name.len();
+    gas_kind[padding..].copy_from_slice(name);
+    max_amount.copy_from_slice(&bound.max_amount.to_be_bytes());
+    max_price.copy_from_slice(&bound.max_price_per_unit.to_be_bytes());
+
+    FieldElement::from_bytes_be(&buffer).expect("Packed resource should fit into felt")
+}
```

Comment:
I've tested to implement this when I've seen the formula landing on the doc. Wondering why do we need the `split_at_must` use here?

Here's what it could be merged to this impl:
```rust
fn encode_gas_bound(name: &[u8], bound: &ResourceBounds) -> FieldElement {
    let mut buffer = [0u8; 32];
    // Name is always 7.5 bytes, so 8 being the limit and we implicitely have the 4 bits not used cleared.
    let padding = 8 - name.len();

    buffer[padding..(8)].copy_from_slice(name);
    // max_amount is u64 so the slice is already good size
    buffer[8..16].copy_from_slice(&bound.max_amount.to_be_bytes());
    // max_price_per_unit is u128 so slice is already good size
    buffer[16..].copy_from_slice(&bound.max_price_per_unit.to_be_bytes());

    FieldElement::from_bytes_be(&buffer).expect("Packed resource should fit into a FieldElement")        
}
```